test/jdk/performance/client/RenderPerfTest/src/renderperf/RenderPerfTest.java

```diff
@@ -602,398 +821,1593 @@ private static BufferedImage makeManagedBI(final int type) {
+        }
+
+        if (help) {
+            help();
+        }
+
+        if (CALIBRATION) {
+            Method m = RenderPerfTest.class.getDeclaredMethod("testCalibration");
+            testCases.add(0, m); // first
+        }
+
+        if (VERBOSE) {
+            System.out.print("##############################################################\n");
+            System.out.printf("# %s\n", VERSION);
+            System.out.print("##############################################################\n");
+            System.out.printf("# Java: %s\n", System.getProperty("java.runtime.version"));
+            System.out.printf("#   VM: %s %s (%s)\n", System.getProperty("java.vm.name"), System.getProperty("java.vm.version"), System.getProperty("java.vm.info"));
+            System.out.printf("#   OS: %s %s (%s)\n", System.getProperty("os.name"), System.getProperty("os.version"), System.getProperty("os.arch"));
+            System.out.printf("# CPUs: %d (virtual)\n", Runtime.getRuntime().availableProcessors());
+            System.out.print("##############################################################\n");
+            System.out.printf("# AWT Toolkit:                %s \n", TOOLKIT.getClass().getSimpleName());
+            System.out.printf("# Execution mode:             %s\n", EXEC_MODE);
+            System.out.printf("# GraphicsConfiguration mode: %s\n", GC_MODE);
+            System.out.print("##############################################################\n");
+            System.out.printf("# Repeats:                    %d\n", REPEATS);
+            System.out.printf("# NW:                         %d\n", NW);
+            System.out.printf("# N:                          %d\n", N);
+            System.out.printf("# WARMUP_COUNT:               %d\n", WARMUP_COUNT);
+            System.out.printf("# Unit:                       %s\n", USE_FPS ? "FPS" : "TIME(ms)");
+            System.out.print("##############################################################\n");
+            System.out.printf("# Font:                      '%s'\n", TEXT_FONT);
+            System.out.printf("# Text: '%s'\n", TEXT_STR);
+            System.out.printf("# FontSize:                   %s\n", TEXT_SIZE_DEFAULT);
+            System.out.printf("# FontSizeLarge:              %s\n", TEXT_SIZE_LARGE);
+            System.out.print("##############################################################\n");
+        }
+
+        // Graphics Configuration handling:
+        final Set<String> fontNames = new LinkedHashSet<>();
+        final Map<String, GraphicsConfiguration> gcByID = new LinkedHashMap<>();
+        final Map<GraphicsConfiguration, String> idByGC = new HashMap<>();
+
+        final GraphicsEnvironment ge = GraphicsEnvironment.getLocalGraphicsEnvironment();
+
+        for (Font f : ge.getAllFonts()) {
+            fontNames.add(f.getName());
+        }
+        // Check font:
+        if (!fontNames.contains(TEXT_FONT)) {
```

Comment:
BTW, I'm getting the following error message with the default font:
Bad font name: [Dialog] !

The check does not work for logical font names like "Dialog"

Reply:
Ok I will test 'generic' font names too